core/src/data/clientGeoJsonSource.cpp

```diff
@@ -77,6 +78,83 @@ ClientGeoJsonSource::ClientGeoJsonSource(std::shared_ptr<Platform> _platform,
 
 ClientGeoJsonSource::~ClientGeoJsonSource() {}
 
+struct add_centroid {
+
+    geometry::point<double>& pt;
```

Comment:
Sorry, missed these. Will address some of these in a followup.

Sure we can, its obvious though? Will change it to be explicit.